### PR TITLE
Replacement for the ietf algs

### PR DESCRIPTION
**Pull request by ericvoit**
Matches the TCG Table.

### CODE REVIEW

File: ietf-tcg-algs@2020-08-26.yang [hunk 1]
```diff
@@ -0,0 +1,692 @@
...
+    if-feature "TPM12 or TPM20";
+    base tpm12;
+    base tpm2;
+    base hash;
+    base symmetric;  
+    description
+      "The XOR encryption algorithm.";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      TCG TPM 2.0 library specification";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:254]:
```suggestion
      TCG TPM 2.0 library specification. ALG_ID: 0x000A";

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 2]
```diff
@@ -0,0 +1,692 @@
...
+
+  identity TPM_ALG_SHA256 {
+    if-feature "TPM20";
+    base tpm2;
+    base hash;
+    description
+      "The SHA 256 algorithm";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      ISO/IEC 10118-3";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:265]:
```suggestion
      ISO/IEC 10118-3. ALG_ID: 0x000B";    

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 3]
```diff
@@ -0,0 +1,692 @@
...
+
+  identity TPM_ALG_SHA384  {
+    if-feature "TPM20";
+    base tpm2;
+    base hash;
+    description
+      "The SHA 384 algorithm";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      ISO/IEC 10118-3";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:276]:
```suggestion
      ISO/IEC 10118-3. ALG_ID: 0x000C";

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 4]
```diff
@@ -0,0 +1,692 @@
...
+
+  identity TPM_ALG_SHA512  {
+    if-feature "TPM20";
+    base tpm2;
+    base hash;
+    description
+      "The SHA 512 algorithm";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      ISO/IEC 10118-3";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:287]:
```suggestion
      ISO/IEC 10118-3. ALG_ID: 0x000D";    

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 5]
```diff
@@ -0,0 +1,692 @@
...
+  }
+
+  identity TPM_ALG_NULL {
+    if-feature "TPM20";
+    base tpm2;
+    description
+      "NULL algorithm";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      TCG TPM 2.0 library specification";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:297]:
```suggestion
      TCG TPM 2.0 library specification. ALG_ID: 0x0010";     

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 6]
```diff
@@ -0,0 +1,692 @@
...
+
+  identity TPM_ALG_SM3_256 {
+    if-feature "TPM20";
+    base tpm2;
+    base hash;
+    description
+      "The SM3 hash algorithm.";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      GM/T 0004-2012 - SM3_256";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:308]:
```suggestion
      GM/T 0004-2012 - SM3_256. ALG_ID: 0x0012";    

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 7]
```diff
@@ -0,0 +1,692 @@
...
+
+  identity TPM_ALG_SM4 {
+    if-feature "TPM20";
+    base tpm2;
+    base symmetric;
+    description
+      "SM4 symmetric block cipher";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      GB/T 32907-2016";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:319]:
```suggestion
      GB/T 32907-2016. ALG_ID: 0x0013";    

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 8]
```diff
@@ -0,0 +1,692 @@
...
+
+  identity TPM_ALG_RSASSA {
+    if-feature "TPM20";
+    base tpm2;
+    base asymmetric;
+    base signing;  
+    description
+      "Signature algorithm defined in section 8.2 (RSASSAPKCS1-v1_5)";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and RFC 8017";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:330]:
```suggestion
      "TCG_Algorithm_Registry_r1p32_pub Table 3 and RFC 8017.
      ALG_ID: 0x0014";    

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 9]
```diff
@@ -0,0 +1,692 @@
...
+
+  identity TPM_ALG_RSAES {
+    if-feature "TPM20";
+    base tpm2;
+    base asymmetric;
+    base encryption-mode;
+    description
+      "Signature algorithm defined in section 7.2 (RSAES-PKCS1-v1_5)";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and RFC 8017";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:341]:
```suggestion
      "TCG_Algorithm_Registry_r1p32_pub Table 3 and RFC 8017
      ALG_ID: 0x0015";

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 10]
```diff
@@ -0,0 +1,692 @@
...
+
+  identity TPM_ALG_RSAPSS {
+    if-feature "TPM20";
+    base tpm2;
+    base asymmetric;
+    base signing;
+    description
+      "Padding algorithm defined in section 8.1 (RSASSA PSS)";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and RFC 8017";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:352]:
```suggestion
      "TCG_Algorithm_Registry_r1p32_pub Table 3 and RFC 8017.
      ALG_ID: 0x0016";   

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 11]
```diff
@@ -0,0 +1,692 @@
...
+
+  identity TPM_ALG_OAEP {
+    if-feature "TPM20";
+    base tpm2;
+    base asymmetric;
+    base encryption-mode;
+    description
+      "Padding algorithm defined in section 7.1 (RSASSA OAEP)";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and RFC 8017";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:363]:
```suggestion
      "TCG_Algorithm_Registry_r1p32_pub Table 3 and RFC 8017.
      ALG_ID: 0x0017";  

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 12]
```diff
@@ -0,0 +1,692 @@
...
+  identity TPM_ALG_ECDSA {
+    if-feature "TPM20";
+    base tpm2;
+    base asymmetric;
+    base signing;
+    description
+      "Signature algorithm using elliptic curve cryptography (ECC)";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      ISO/IEC 14888-3";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:375]:
```suggestion
      ISO/IEC 14888-3. ALG_ID: 0x0018";    

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 13]
```diff
@@ -0,0 +1,692 @@
...
+  identity TPM_ALG_ECDH {
+    if-feature "TPM20";
+    base tpm2;
+    base asymmetric;
+    base method;
+    description
+      "Secret sharing using ECC";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      NIST SP800-56A and RFC 7748.";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:387]:
```suggestion
      NIST SP800-56A and RFC 7748. ALG_ID: 0x0019";    

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 14]
```diff
@@ -0,0 +1,692 @@
...
+    if-feature "TPM20";
+    base tpm2;
+    base asymmetric;
+    base signing;
+    base anonymous-signing;
+    description
+      "Elliptic-curve based anonymous signing scheme";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      TCG TPM 2.0 library specification";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:400]:
```suggestion
      TCG TPM 2.0 library specification. ALG_ID: 0x001A";   

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 15]
```diff
@@ -0,0 +1,692 @@
...
+    base encryption-mode;
+    base method;
+    description
+      "SM2 – depending on context, either an elliptic-curve based,
+      signature algorithm, an encryption scheme, or a key exchange 
+      protocol";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      A GM/T 0003.1–2012, GM/T 0003.2–2012, GM/T 0003.3–2012,
+      GM/T 0003.5–2012   SM2";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:417]:
```suggestion
      GM/T 0003.5–2012 SM2. ALG_ID: 0x001B";    

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 16]
```diff
@@ -0,0 +1,692 @@
...
+  identity TPM_ALG_ECSCHNORR {
+    if-feature "TPM20";
+    base tpm2;
+    base asymmetric;
+    base signing;
+    description
+      "Elliptic-curve based Schnorr signature";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      TCG TPM 2.0 library specification";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:429]:
```suggestion
      TCG TPM 2.0 library specification. ALG_ID: 0x001C";       

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 17]
```diff
@@ -0,0 +1,692 @@
...
+  identity TPM_ALG_ECMQV {
+    if-feature "TPM20";
+    base tpm2;
+    base asymmetric;
+    base method;
+    description
+      "Two-phase elliptic-curve key";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      NIST SP800-56A";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:441]:
```suggestion
      NIST SP800-56A. ALG_ID: 0x001D";    

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 18]
```diff
@@ -0,0 +1,692 @@
...
+  identity TPM_ALG_KDF1_SP800_56A {
+    if-feature "TPM20";
+    base tpm2;
+    base hash;
+    base method;
+    description
+      "Concatenation key derivation function";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      NIST SP800-56A  (approved alternative1) section 5.8.1";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:453]:
```suggestion
      NIST SP800-56A  (approved alternative1) section 5.8.1.
      ALG_ID: 0x0020";    

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 19]
```diff
@@ -0,0 +1,692 @@
...
+  identity TPM_ALG_KDF2  {
+    if-feature "TPM20";
+    base tpm2;
+    base hash;
+    base method;
+    description
+      "Key derivation function";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      IEEE 1363a-2004 KDF2 section 13.2";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:465]:
```suggestion
      IEEE 1363a-2004 KDF2 section 13.2. ALG_ID: 0x0021";    

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 20]
```diff
@@ -0,0 +1,692 @@
...
+      IEEE 1363a-2004 KDF2 section 13.2";    
+  }
+
+  identity TPM_ALG_KDF1_SP800_108 {
+    base TPM_ALG_KDF2;
+    description
+      "A key derivation method";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      NIST SP800-108 - Section 5.1 KDF";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:474]:
```suggestion
      NIST SP800-108 - Section 5.1 KDF. ALG_ID: 0x0022";    

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 21]
```diff
@@ -0,0 +1,692 @@
...
+  identity TPM_ALG_ECC {
+    if-feature "TPM20";
+    base tpm2;
+    base asymmetric;
+    base object-type;
+    description
+      "Prime field ECC";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      ISO/IEC 15946-1";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:486]:
```suggestion
      ISO/IEC 15946-1. ALG_ID: 0x0023";    

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 22]
```diff
@@ -0,0 +1,692 @@
...
+  }
+
+  identity TPM_ALG_SYMCIPHER {
+    if-feature "TPM20";
+    base tpm2;
+    description
+      "Object type for a symmetric block cipher";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      TCG TPM 2.0 library specification";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:496]:
```suggestion
      TCG TPM 2.0 library specification. ALG_ID: 0x0025";          

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 23]
```diff
@@ -0,0 +1,692 @@
...
+
+  identity TPM_ALG_CAMELLIA {
+    if-feature "TPM20";
+    base tpm2;
+    base symmetric;
+    description
+      "The Camellia algorithm";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      ISO/IEC 18033-3";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:507]:
```suggestion
      ISO/IEC 18033-3. ALG_ID: 0x0026";    

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 24]
```diff
@@ -0,0 +1,692 @@
...
+
+  identity TPM_ALG_SHA3_256  {
+    if-feature "TPM20";
+    base tpm2;
+    base hash;
+    description
+      "ISO/IEC 10118-3 - the SHA 256 algorithm";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      NIST PUB FIPS 202";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:518]:
```suggestion
      NIST PUB FIPS 202. ALG_ID: 0x0027";          

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 25]
```diff
@@ -0,0 +1,692 @@
...
+
+  identity TPM_ALG_SHA3_384 {
+    if-feature "TPM20";
+    base tpm2;
+    base hash;
+    description
+      "The SHA 384 algorithm";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      NIST PUB FIPS 202";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:529]:
```suggestion
      NIST PUB FIPS 202. ALG_ID: 0x0028";   

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 26]
```diff
@@ -0,0 +1,692 @@
...
+
+  identity TPM_ALG_SHA3_512 {
+    if-feature "TPM20";
+    base tpm2;
+    base hash;
+    description
+      "The SHA 512 algorithm";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      NIST PUB FIPS 202";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:540]:
```suggestion
      NIST PUB FIPS 202. ALG_ID: 0x0029";   

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 27]
```diff
@@ -0,0 +1,692 @@
...
+  identity TPM_ALG_CMAC {
+    if-feature "TPM20";
+    base tpm2;
+    base symmetric;
+    base signing;
+    description
+      "block Cipher-based Message Authentication Code (CMAC)";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      ISO/IEC 9797-1:2011 Algorithm 5";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:552]:
```suggestion
      ISO/IEC 9797-1:2011 Algorithm 5. ALG_ID: 0x003F";   

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 28]
```diff
@@ -0,0 +1,692 @@
...
+  identity TPM_ALG_CTR {
+    if-feature "TPM20";
+    base tpm2;
+    base symmetric;
+    base encryption-mode;
+    description
+      "Counter mode";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      ISO/IEC 10116";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:564]:
```suggestion
      ISO/IEC 10116. ALG_ID: 0x0040";    

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 29]
```diff
@@ -0,0 +1,692 @@
...
+
+  identity TPM_ALG_OFB {
+    base tpm2;
+    base symmetric;
+    base encryption-mode;
+    description
+      "Output Feedback mode";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      ISO/IEC 10116";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:575]:
```suggestion
      ISO/IEC 10116. ALG_ID: 0x0041";    

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 30]
```diff
@@ -0,0 +1,692 @@
...
+  identity TPM_ALG_CBC {
+    if-feature "TPM20";
+    base tpm2;
+    base symmetric;
+    base encryption-mode;
+    description
+      "Cipher Block Chaining mode";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      ISO/IEC 10116";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:587]:
```suggestion
      ISO/IEC 10116. ALG_ID: 0x0042";    

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 31]
```diff
@@ -0,0 +1,692 @@
...
+  identity TPM_ALG_CFB {
+    if-feature "TPM20";
+    base tpm2;
+    base symmetric;
+    base encryption-mode;
+    description
+      "Cipher Feedback mode";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      ISO/IEC 10116";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:599]:
```suggestion
      ISO/IEC 10116. ALG_ID: 0x0043";    

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 32]
```diff
@@ -0,0 +1,692 @@
...
+  identity TPM_ALG_ECB  {
+    if-feature "TPM20";
+    base tpm2;
+    base symmetric;
+    base encryption-mode;
+    description
+      "Electronic Codebook mode";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      ISO/IEC 10116";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:611]:
```suggestion
      ISO/IEC 10116. ALG_ID: 0x0044";    

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 33]
```diff
@@ -0,0 +1,692 @@
...
+    base tpm2;
+    base symmetric;
+    base signing;
+    base encryption-mode;
+    description
+      "Counter with Cipher Block Chaining-Message Authentication 
+      Code (CCM)";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      NIST SP800-38C";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:625]:
```suggestion
      NIST SP800-38C. ALG_ID: 0x0050";    

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 34]
```diff
@@ -0,0 +1,692 @@
...
+    if-feature "TPM20";
+    base tpm2;
+    base symmetric;
+    base signing;
+    base encryption-mode;
+    description
+      "Galois/Counter Mode (GCM)";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      NIST SP800-38D";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:638]:
```suggestion
      NIST SP800-38D. ALG_ID: 0x0051";    

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 35]
```diff
@@ -0,0 +1,692 @@
...
+    if-feature "TPM20";
+    base tpm2;
+    base symmetric;
+    base signing;
+    base encryption-mode;
+    description
+      "AES Key Wrap (KW)";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      NIST SP800-38F";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:651]:
```suggestion
      NIST SP800-38F. ALG_ID: 0x0052";    

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 36]
```diff
@@ -0,0 +1,692 @@
...
+    if-feature "TPM20";
+    base tpm2;
+    base symmetric;
+    base signing;
+    base encryption-mode;
+    description
+      "AES Key Wrap with Padding (KWP)";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      NIST SP800-38F";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:664]:
```suggestion
      NIST SP800-38F. ALG_ID: 0x0053";    

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 37]
```diff
@@ -0,0 +1,692 @@
...
+    if-feature "TPM20";
+    base tpm2;
+    base symmetric;
+    base signing;
+    base encryption-mode;
+    description
+      "Authenticated-Encryption Mode";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      NIST SP800-38F";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:677]:
```suggestion
      NIST SP800-38F. ALG_ID: 0x0054";    

```

File: ietf-tcg-algs@2020-08-26.yang [hunk 38]
```diff
@@ -0,0 +1,692 @@
...
+  identity TPM_ALG_EDDSA {
+    if-feature "TPM20";
+    base tpm2;
+    base asymmetric;
+    base signing;
+    description
+      "Edwards-curve Digital Signature Algorithm (PureEdDSA)";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and
+      RFC 8032";
```
Comment (henkbirkholz) [inline ietf-tcg-algs@2020-08-26.yang:689]:
```suggestion
      RFC 8032. ALG_ID: 0x0060";    

```

**Review by henkbirkholz — APPROVED** [review]
I really like the new identity mapping. For direct mapping to TLV'ish/binary representation, I added the actual al-ids to the reference statements in each identity.

I also really like the small taxonomy including multi parents for ease of use and selection. In general, this will become its own work, so updates to the registry. Question is: does it therefore make sense to split now already?

File: ietf-tcg-algs@2020-08-26.yang [hunk 39]
```diff
@@ -0,0 +1,692 @@
...
+    if-feature "TPM12 or TPM20";
+    base tpm12;  
+    base tpm2;
+    base asymmetric;
+    base object-type;
+    description
+      "RSA algorithm";
+    reference  
+      "TCG_Algorithm_Registry_r1p32_pub Table 3 and 
+      RFC 8017";
```
[the remark this answers is not in the document]
Comment (ericvoit) [reply ietf-tcg-algs@2020-08-26.yang:163]: I like the additions of the ALG_ID

**Review by ericvoit — COMMENTED** [review]
I like the additions of the ALG_ID

**Pull request merged**